src/main/java/io/localmotion/picture/repository/PictureRepository.java

```diff
@@ -0,0 +1,21 @@
+package io.localmotion.picture.repository;
+
+import io.localmotion.picture.entity.Picture;
+import io.localmotion.picture.entity.PictureId;
+
+import java.util.List;
+
+public interface PictureRepository {
+
+	Picture findPictureByPictureId(PictureId pictureId);
+
+	Picture uploadPicture(Picture picture);
+
+	Picture updatePicture(Picture picture);
+
+	void removePicture(String userId, String pictureId);
+
+	Picture findPictureByImageId(String imageId);
+
+	List<Picture> findPictureById(String id);
```

Comment:
For both implementation and interface: we just need the find, store and delete all keyed on the pictureId, which should be a uuid and not the userid (the userid that did the upload is just an attribute not a key)